### PR TITLE
chore: Add retry logic for the ci workflows

### PR DESCRIPTION
**Pull request by lqiu96**
Add retry logic for maven install to fix the random issues pulling maven dependencies from maven central.

### CODE REVIEW

File: .kokoro/build.sh [hunk 1]
```diff
@@ -92,58 +104,51 @@ case ${JOB_TYPE} in
     fi
     ;;
   samples)
```
Comment (meltsufin) [inline .kokoro/build.sh:106]: You can delete this whole samples case. It's dead code.

Comment (lqiu96) [reply .kokoro/build.sh:106]: Yeah, the samples aren't testing anything right now. I thought we were fine with keeping this in preparation for the handwritten libraries moving into the monorepo. Are we going to come up with different way to test samples for handwritten libraries and I should just remove this block?

Comment (meltsufin) [reply .kokoro/build.sh:106]: Yeah, I think I said it's fine to keep, but seeing how complex the code is, I'd rather just get rid of it. We won't have to deal with handwritten samples in the future.

Comment (lqiu96) [reply .kokoro/build.sh:106]: Ok. I'll remove it.

Comment (lqiu96) [reply .kokoro/build.sh]: I've kept the samples.cfg for the kokoro job in for now. It just currently does nothing -- I'm considering deleting this config  and in g3 so it doesn't waste resources in our pool.

**Comment by lqiu96** [comment]
@meltsufin Could you take another look when you get a chance?

**Review by meltsufin — APPROVED** [review]

**Pull request merged**